starknet_py/contract.py

```diff
@@ -99,8 +98,9 @@ async def wait_for_acceptance(
 @add_sync_methods
 @dataclass(frozen=True)
 class InvokeResult(SentTransaction):
-    contract: ContractData = None
-    invoke_transaction: InvokeFunction = None
+    # We ensure these not None in __post_init__
+    contract: ContractData = None  # pyright: ignore
+    invoke_transaction: InvokeFunction = None  # pyright: ignore
```

Comment:
This is not validated in `__post_init__`

Reply:
Yes is is
```
    def __post_init__(self):
        assert self.contract is not None
        assert self.invoke_transaction is not None
```